explicitly provide memory format when calling to *_like operators (Redo of 81bf7364)

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30081 [WIP] Modify default memory format
* #30082 [WIP] Tests workaround
* #30008 explicitly provide memory format when calling to *_like operators
* #30007 explicitly provide memory format when calling to *_like operators
* **#30006 explicitly provide memory format when calling to *_like operators (Redo of 81bf7364)**
* #30005 explicitly provide memory format when calling to *_like operators (Redo of cc1c01)
* #30004 explicitly provide memory format when calling to *_like operators (Redo of e3e06549)
* #30003 explicitly provide memory format when calling to *_like operators (Redo of 4b4aa)
* #30002 explicitly provide memory format when calling to *_like operators (Redo of ce438f6967)
* #30001 explicitly provide memory format when calling to *_like operators (Redo of 631b22d)
* #30000 explicitly provide memory format when calling to *_like operators

Differential Revision: [D18575984](https://our.internmc.facebook.com/intern/diff/D18575984)

VitalyFedyunin committed:
explicitly provide memory format when calling to *_like operators (Redo of 81bf7364)

[ghstack-poisoned]

mrshenli reviewed (approved):
Approving changes on `dist_engine.cpp`

VitalyFedyunin committed:
update tests on "explicitly provide memory format when calling to *_like operators (Redo of 81bf7364)"


Differential Revision: [D18575984](https://our.internmc.facebook.com/intern/diff/D18575984)

[ghstack-poisoned]
